server/contexts/authz/authz.go

```diff
@@ -22,12 +22,24 @@ func FromContext(ctx context.Context) (*AuthorizationContext, bool) {
 	return v, ok
 }
 
+// AuthenticationMethod identifies the method used to authenticate.
+type AuthenticationMethod int
+
+// List of supported authentication methods.
+const (
+	AuthnUserToken AuthenticationMethod = iota
```

Comment:
Let's add a brief comment about the three types of entities that authenticate to Fleet, `User`, `Host` and `Device` are not very descriptive :)